### PR TITLE
Fix some embedded shell issues and add tests

### PR DESCRIPTION
**Pull request by stefwalter**
* The shell can be embedded as a component that configures a single host.
 * So remove the navbar, URL mangling, and other  behavior when doing so.
 * Don't display the dashboard by default in these cases either.
 * Add test for this stuff.

### CODE REVIEW

**Comment by stefwalter** [comment]
Test failures are unrelated. Filed #2810 and #2811

**Comment by stefwalter** [comment]
Rebased.

**Comment by stefwalter** [comment]
Test failure unrelated. Tracked at #2835

File: test/check-embed [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import os
+
+class TestEmbed(MachineCase):
+    def testBasic(self):
+        b = self.browser
+        m = self.machine
+
+        self.start_cockpit()
+
+        # Should log in automatically due to above
```
Comment (mvollmer) [inline test/check-embed]: Is this comment accurate?  There is no automatic login going on here, I think.

Comment (stefwalter) [inline test/check-embed]: Thanks. Fixed.

**Comment by mvollmer** [comment]
Verified locally.

**Pull request closed without merging**